src/simulation/dynamics/constraintEffector/constraintDynamicEffector.rst

```diff
@@ -0,0 +1,80 @@
+
+Executive Summary
+-----------------
+
+The constraint effector class is used to define a physical connection between two separate 
+spacecraft. This class takes in the connection location on each spacecraft in their respective 
+body frames as well as a relative vector between these two connection points. It also takes in 
+two gain tuning parameters to tailor the stiffness and damping of the connection between the 
+two spacecraft.
+
+The constraint effector class is an instantiation of the dynamic effector abstract class. The 
+included test is validating the interaction between two spacecraft rigid body hubs that are 
+attached through a constraint effector. In this case, two identical spacecraft are connected 
+by a 0.1 meter long arm which is enforced with high stiffness and damping to be virtually rigid.
+
+Detailed Module Description
+---------------------------
+
+A constraint effector is a combination of two separate 3-degree-of-freedom holonomic constraints: a 
+direction constraint which enforces the position of the connection point on each spacecraft 
+relative to the other, and an attitude constraint which enforces the attitude of each spacecraft
+relative to the other. The constraint effector works by correcting violations of these constraints 
+as well as their derivatives.
+
+Mathematical Modeling
+^^^^^^^^^^^^^^^^^^^^^
+See the following conference paper for a detailed description of the mathematical model.
+
+.. note::
+
+    J. Vaz Carneiro, A. Morell and H. Schaub, `"Post-Docking Complex 
+    Spacecraft Dynamics Using Baumgarte Stabilization" <https://hanspeterschaub.info/Papers/VazCarneiro2023b.pdf>`_,
+    AAS/AIAA Astrodynamics Specialist Conference, Big Sky, Montana, Aug. 13-17, 2023
+
+User Guide
+----------
+This section outlines the steps needed to setup a Constraint Dynamic Effector in Python using Basilisk.
+
+#. Import the constraintDynamicEffector class::
+
+    from Basilisk.simulation import constraintDynamicEffector
+
+#. Create an instantiation of a holonomic constraint::
+
+    constraintEffector = constraintDynamicEffector.ConstraintDynamicEffector()
+
+#. Define all physical parameters for the constraint::
```

Comment:
update module documentation to talk about using setter and getter functions to set and read user configurable module variables